src/docs/arch-cursor.dox

```diff
@@ -3,13 +3,393 @@
+
+Incremental backups work much the same way, except that the file list is reduced to files that
+have changed since a previous backup referenced in the configuration when the cursor is opened.
+The other twist is that for each file returned, the caller does a duplicate operation on the backup
+cursor, and the duplicate code actually returns a specialized incremental backup cursor.  This kind
+of cursor has its own \c next method that causes it to return information about individual pieces
+of this file that need to be copied.  The code to implement incremental cursors
+is in \c src/cursor/cur_backup_incr.c .
+
+@section arch_cursor_join Join cursors
+
+Join cursors implement a join mechanism for WiredTiger.  The idea is that joins can be configured
+by opening a special join cursor on a table, and attaching the table's index cursors to it, to
+return rows that match a filter, like:
+```
+(row.price > 100 and row.price <= 200) and row.in_stock < 10
+```
+Building up the conditions essentially creates a tree that is used for evaluation.  This is stored
+in \c WT_CURSOR_JOIN, and is returned as the cursor object.  Entries in the tree representing
+an index's participation in one clause are stored in \c WT_CURSOR_JOIN_ENTRY objects.  \c WT_CURSOR_JOIN and
+\c WT_CURSOR_JOIN_ENTRY objects can be composed in a hierarchical manner, representing the shape of the tree representing
+the query.  These two structures are somewhat static, being created when the join is created.
+
+The join cursor contains a pointer to a \c WT_CURSOR_JOIN_ITER, which in some sense encodes the "position"
+of the cursor.  To get the next row that satisfies a join requires that multiple cursors be iterated.
+Generally, the "left-most" index cursor is iterated first.  Using the example above, the \c price
+index would be iterated, skipping over any entries that did not satisfy `(row.price > 100 and row.price <= 200)`.
+Then other cursors are checked to see that any other conditions are satisfied.
+The join may be configured to use Bloom filters, and when that occurs,
+\c WT_CURSOR_JOIN_ENTRIES contain the bloom filter for individual index checks.  This allows checks
+to occur quickly, at the expense of an initialization that occurs when the cursor begins iteration.
+
+During a join cursor iteration, multiple \c WT_CURSOR_JOIN_ITER objects may be created.
+There is one \c WT_CURSOR_JOIN_ITER object corresponding to each level of nesting (queries can have
+arbitrary nesting of \c AND/OR).  Each \c WT_CURSOR_JOIN_ITER allows the question to be asked: does
+the current position of the main table cursor satisfy the join conditions for this part of the join tree?
+
+Part of the reason for this dynamic structure is that disjunctions may require some dynamic action.
+Consider
+```
+(row.price > 100 and row.in_stock < 10) .... or (row.aisle == 12 and row.on_sale == 1)
+```
+The last part of the query (being part of an OR clause) may be first executed after a number of items
```

Comment:
may be -> will probably be

Reply:
Changed this from "may be" to "will be".